app/controllers/questions_controller.rb

```diff
@@ -104,9 +104,30 @@ def destroy
     flash[:success] = "You have successfully deleted the question"    
   end
 
+#  def up_vote
+#   @question=Question.find(params[:id])
+#  @question.up_vote = @question.up_vote + 1
+#  end
+#  def down_vote
+#    @question=Question.find(params[:id])
+#    @question.down_vote = @question.down_vote + 1
+#  end
+
+
+  def up_vote
+    @question = Question.find(params[:id])
+    @question.liked_by current_user
+
+    if request.xhr?
+      render json: { count: @question.get_likes.size, id: params[:id] }
```

Comment:
where is the method `get_likes` defined? I couldn't see it in the `Question` model.